[quant] Fusion support for conv1d + ReLU

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38452 [quant] Support for fused ConvBn1d and ConvBnRelu1d modules
* #38449 [quant] Support for functional quantized::conv1d
* #38441 [quant][graphmode] Add support for quantized conv1d + relu fusion
* **#38438 [quant] Fusion support for conv1d + ReLU**
* #38407 [quant] Return default qconfig when backend is 'none'

Summary:
Fusion for PTQ flow in eager mode. Graph mode to follow

Test Plan:
python test/test_quantization.py TestFusion

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21575920](https://our.internmc.facebook.com/intern/diff/D21575920)

supriyar committed:
[quant] Fusion support for conv1d + ReLU
Summary:
Fusion for PTQ flow in eager mode. Graph mode to follow

Test Plan:
python test/test_quantization.py TestFusion

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on torch/quantization/fuse_modules.py:
```diff
@@ -97,6 +97,7 @@ def fuse_known_modules(mod_list):
         (torch.nn.Conv2d, torch.nn.BatchNorm2d, torch.nn.ReLU): fuse_conv_bn_relu,
         (torch.nn.Conv3d, torch.nn.BatchNorm3d): fuse_conv_bn,
         (torch.nn.Conv3d, torch.nn.BatchNorm3d, torch.nn.ReLU): fuse_conv_bn_relu,
+        (torch.nn.Conv1d, torch.nn.ReLU): torch.nn.intrinsic.ConvReLU1d,
```
Please also add support for Conv1d + BatchNorm1d and Conv1d+BatchNorm1d+Relu for eval, QAT support can be a TODO

supriyar committed:
Update on "[quant] Fusion support for conv1d + ReLU"


Summary:
Fusion for PTQ flow in eager mode. Graph mode to follow

Test Plan:
python test/test_quantization.py TestFusion

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Fusion support for conv1d + ReLU"


Summary:
Fusion for PTQ flow in eager mode. Graph mode to follow

Test Plan:
python test/test_quantization.py TestFusion

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
